diff -ur x264.h x264.h
--- x264.h	2024-11-27 17:12:17.931362486 +0100
+++ x264.h	2024-11-27 23:33:12.758618511 +0100
@@ -617,6 +617,11 @@
 
     /* For internal use only */
     void *opaque;
+
+    /*== Irdeto specific parameters ==*/
+    int b_irdeto_extension;
+    int pps_id;
+
 } x264_param_t;
 
 X264_API void x264_nal_encode( x264_t *h, uint8_t *dst, x264_nal_t *nal );
diff -ur Makefile Makefile
--- Makefile	2024-11-27 17:12:17.806432600 +0100
+++ Makefile	2024-11-27 23:47:28.745176530 +0100
@@ -2,12 +2,12 @@
 
 include config.mak
 
-vpath %.c $(SRCPATH)
-vpath %.h $(SRCPATH)
-vpath %.S $(SRCPATH)
-vpath %.asm $(SRCPATH)
-vpath %.rc $(SRCPATH)
-vpath %.manifest $(SRCPATH)
+# vpath %.c $(SRCPATH)
+# vpath %.h $(SRCPATH)
+# vpath %.S $(SRCPATH)
+# vpath %.asm $(SRCPATH)
+# vpath %.rc $(SRCPATH)
+# vpath %.manifest $(SRCPATH)
 
 CFLAGS += $(CFLAGSPROF)
 LDFLAGS += $(LDFLAGSPROF)
